securesystemslib/signer/_aws_signer.py

```diff
@@ -206,18 +229,20 @@ def _get_hash_algorithm(public_key: Key) -> str:
     def sign(self, payload: bytes) -> Signature:
         """Sign the payload with the AWS KMS key
 
+        This method sends the payload to AWS KMS, where it is signed using the specified
+        key and algorithm using the raw message type.
```

Comment:
Good addition, thanks: I had not realized this is not prehashed.

This is not for this PR but just as a note:
* Users probably expect prehashing (instead of sending potentially large files to AWS)
* Confusingly this method does prehash the content... the result is just not used

Reply:
> Confusingly this method does prehash the content... the result is just not used

If you are not sending the prehashed content, I suggest to  remove the code that hashes the content. This means you no longer need `self.hash_algorithm`, which means that you no longer need `self._get_aws_signing_algo` (except maybe for catching unsupported schemes).